database/db.go

```diff
@@ -128,6 +154,38 @@ func (db *DB) GetUser(uuid string) (User, error) {
 	return user, err
 }
 
+func (db *DB) GetUserByUUID(uuid []string) ([]User, error) {
```

Comment:
I think the looping behaviour should belong to the HTTP handler, not the DB layer.
```
func GetUsersHandler(){
    //...
    var users []db.User
    for _, uuid := range uuids {
        usr, err := GetUser(uuid)
        if err != nil {
            return err
        }
        users = append(users, usr)
    }
}
```

Reply:
I think this should be named `GetUsersByUUID` or just `GetUsers` rather than GetUser
Where the first argument is called `uuids []string`

---

Getting multiple users can be done in a single query using:

This simplifies to something like
```
users := []User{}

// Generate a string like "(?,?)" for as many UUIDs as we are given
fragment := strings.Join(
  strings.Split(strings.Repeat("?", len(uuids)), ""), ",",
)

rows, err := db.conn.Query(fmt.Sprintf(
  `SELECT uuid, email FROM users WHERE uuid IN (%s)`,
  fragment,
))

if err != nil {
  return users, err
}

defer rows.Close()

for rows.Next() {
  var user User
  err := rows.Scan(&user.UUID, &user.Email)
  if err != nil {
    return users, err
  }
  users = append(users, user)
}

return users, nil
```

Alternatively `pq` driver instead of `sql` lets us use a string slice